### PR TITLE
Fix throttling issue in websocket APIs when using JWT tokens

### PR DESCRIPTION
**Pull request by ChamodDamitha**
Fixes https://github.com/wso2/product-apim/issues/8718

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java [hunk 1]
```diff
@@ -322,64 +322,34 @@ public AuthenticationContext authenticateForWebSocket(SignedJWT jwtToken, String
...
-                            }
-                        }
-                    } else {
-                        log.debug("User is NOT authorized to access the Resource. API Subscription validation failed.");
-                        throw new APISecurityException(apiKeyValidationInfoDTO.getValidationStatus(),
-                                "User is NOT authorized to access the Resource. API Subscription validation failed.");
-                    }
-                }
-                log.debug("Ignored subscription validation");
+            log.debug("Begin subscription validation via Key Manager");
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java:325]: can we append the km name here?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java [hunk 2]
```diff
@@ -322,64 +322,34 @@ public AuthenticationContext authenticateForWebSocket(SignedJWT jwtToken, String
...
+            APIKeyValidationInfoDTO apiKeyValidationInfoDTO = validateSubscriptionUsingKeyManager(apiContext,
+                    apiVersion, jwtValidationInfo);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Subscription validation via Key Manager. Status: " +
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java:330]: here as well

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java [hunk 3]
```diff
@@ -322,64 +322,34 @@ public AuthenticationContext authenticateForWebSocket(SignedJWT jwtToken, String
...
             }
-            log.debug("JWT authentication successful.");
-            String endUserToken = null;
-            try {
-                if (jwtGenerationEnabled) {
-                    JWTInfoDto jwtInfoDto =
-                            GatewayUtils.generateJWTInfoDto(jwtValidationInfo, api, null, apiContext, apiVersion);
-                    endUserToken = generateAndRetrieveJWTToken(tokenSignature, jwtInfoDto);
+            if (apiKeyValidationInfoDTO.isAuthorized()) {
+                log.debug("JWT authentication successful.");
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java:334]: can we append the username/any other info here?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java [hunk 4]
```diff
@@ -322,64 +322,34 @@ public AuthenticationContext authenticateForWebSocket(SignedJWT jwtToken, String
...
                 }
-                return GatewayUtils
-                        .generateAuthenticationContext(tokenSignature, jwtValidationInfo, api, null,
-                                getApiLevelPolicy(),
-                                endUserToken, true);
-            } catch (ParseException e) {
-                throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR,
-                        APISecurityConstants.API_AUTH_GENERAL_ERROR_MESSAGE);
+            } else {
+                log.debug("User is NOT authorized to access the Resource. API Subscription validation failed.");
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java:350]: we can use a constant for the error message and reuse in the exception

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java (hunk 1 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/8999/commits/83d8083bf4fdd8144025e55cdf9c2c0259d88850

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java (hunk 2 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/8999/commits/83d8083bf4fdd8144025e55cdf9c2c0259d88850

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java (hunk 3 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/8999/commits/83d8083bf4fdd8144025e55cdf9c2c0259d88850

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java (hunk 4 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/8999/commits/83d8083bf4fdd8144025e55cdf9c2c0259d88850

**Pull request merged**